mlflow/types/llm.py

```diff
@@ -90,24 +127,73 @@ def from_dict(cls, data):
         return cls(**filtered_data)
 
 
+@dataclass
+class FunctionToolCallArguments(_BaseDataclass):
+    """
+    The arguments of a function tool call made by the model.
+
+    Args:
+        arguments (str): A JSON string of arguments that should be passed to the tool.
+        name (str): The name of the tool that is being called.
+    """
+
+    name: str
+    arguments: str
+
+    def __post_init__(self):
+        self._validate_field("name", str, True)
+        self._validate_field("arguments", str, True)
+
+    def to_tool_call(self, id):
+        return ToolCall(id=id, function=self)
+
+
+@dataclass
+class ToolCall(_BaseDataclass):
+    """
+    A tool call made by the model.
+
+    Args:
+        id (str): The ID of the tool call.
+        function (:py:class:`FunctionToolCallArguments`): The arguments of the function tool call.
+        type (str): The type of the object. Currently only "function" is supported.
```

Comment:
As mentioned above, right now this class is just necessary to conform to the OpenAI spec, see the [API spec](https://platform.openai.com/docs/api-reference/chat/object), `choices > message > tool_calls`

<img width="589" alt="Screenshot 2024-09-19 at 1 01 20 PM" src="https://github.com/user-attachments/assets/3dda1b0e-b1d7-4e7a-88da-6f9d8eb01bc1">

In the future, I imagine that they may add other types besides `function`, and then this class will have more utility.